src/main/java/com/blamejared/crafttweaker/impl/loot/conditions/LootConditionManager.java

```diff
@@ -0,0 +1,115 @@
+package com.blamejared.crafttweaker.impl.loot.conditions;
+
+import com.blamejared.crafttweaker.api.CraftTweakerAPI;
+import com.blamejared.crafttweaker.api.CraftTweakerRegistry;
+import com.mojang.datafixers.util.Pair;
+import cpw.mods.modlauncher.api.LamdbaExceptionUtils;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public final class LootConditionManager {
+    private static final Map<Class<? extends ILootConditionTypeBuilder>, Function<CTLootConditionBuilder, ? extends ILootConditionTypeBuilder>> BUILDERS = new HashMap<>();
+    
+    public static void handleBuilderRegistration() {
+        BUILDERS.clear();
+        CraftTweakerRegistry.getZenClassRegistry()
+                .getImplementationsOf(ILootConditionTypeBuilder.class)
+                .stream()
+                .map(LootConditionManager::make)
+                .filter(Objects::nonNull)
+                .forEach(LootConditionManager::register);
+    }
+    
+    @SuppressWarnings("unchecked")
+    public static <T extends ILootConditionTypeBuilder> T get(final CTLootConditionBuilder builder, final Class<T> token) {
+        return (T) Objects.requireNonNull(BUILDERS.get(token), () -> "No loot condition builder for type '" + token.getName() + "' was registered!").apply(builder);
+    }
+    
+    private static <T extends ILootConditionTypeBuilder> Pair<Class<T>, Function<CTLootConditionBuilder, T>> make(final Class<T> clazz) {
+        final Field instanceField = findInstanceField(clazz);
+        final Pair<Constructor<T>, Constructor<T>> constructors = findConstructors(clazz);
+        
+        final boolean hasInstance = instanceField != null;
+        final boolean hasConstructor = constructors.getFirst() != null || constructors.getSecond() != null;
+        final boolean hasBothConstructors = hasConstructor && (constructors.getFirst() != null && constructors.getSecond() != null);
```

Comment:
the `hasConstructor` part is unnecessary, since the subsequent checks cover it.

Reply:
True, although I wanted to avoid the getter call if there was no constructor in the first place, though this is probably premature optimization and does more harm than good. I'll drop the initial check.